pkg/synthetictests/duplicated_events.go

```diff
@@ -250,6 +251,16 @@ func testDuplicatedEventForUpgrade(events monitorapi.Intervals, kubeClientConfig
 }
 
 func testDuplicatedEventForStableSystem(events monitorapi.Intervals, clientConfig *rest.Config, testSuite string) []*junitapi.JUnitTestCase {
+	cfgClient := configclient.NewForConfigOrDie(clientConfig)
+	controlplaneTopology, err := exutil.GetControlPlaneTopologyFromConfigClient(cfgClient.ConfigV1())
+	if err != nil {
+		panic(fmt.Errorf("failed to determine controlplane toplogy: %w", err))
+	}
+
+	// There is no etcd cluster operator when the ControlPlaneTopology is External
+	if *controlplaneTopology == v1.ExternalTopologyMode {
+		return nil
```

Comment:
It's not immediately clear to me that returning nil and appending to the slice of pointers will work, looks like it might, but I'd suggest just returning an empty array. However I think we're actually bypassing the testing later in this func, which may applicable to hypershift. I'd just try to skip the etcdAllowance portion when we detect this, and let the later checks still run.

Reply:
i agree returning an empty slice seems better, assuming the rest of the tests even should be bypassed

Reply:
(and then the nil check later needs to become a length check instead, which probably always should have been there)